include/stdx/ct_string.hpp

```diff
@@ -142,4 +142,11 @@ template <bool B> constexpr auto ct_check = ct_check_t<B>{};
 } // namespace v1
 } // namespace stdx
 
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define STATIC_ASSERT(cond, ...)                                               \
+    []<auto B = cond>() -> bool {                                              \
+        stdx::ct_check<B>.template emit<__VA_ARGS__>();                        \
+        return B;                                                              \
+    }()
```

Comment:
suggestion: integrate `ct_format` in this macro so that it is easy to leverage the real power of `ct_check`. the new macro signature would be something like:

```c++
#define STATIC_ASSERT(cond, msg, ...) \
```

where the macro var args are the arguments to be formatted into `msg`.

Reply:
:+1:  I had the same idea...